templates/base/client/.hathora/client.ts.hbs

```diff
@@ -57,31 +60,12 @@ export class HathoraClient {
     token: string,
     stateId: StateId,
     onUpdate?: UpdateCallback,
-    onError?: ErrorCallback
+    onError?: ErrorCallback,
+    transportType: TransportType = TransportType.WebSocket
   ): Promise<HathoraConnection> {
-    return new Promise((resolve, reject) => {
-      const socket = new WebSocket(`wss://${COORDINATOR_HOST}/${this.appId}`);
-      socket.binaryType = "arraybuffer";
-      socket.onclose = (e) => reject(transformCoordinatorFailure(e));
-      socket.onopen = () =>
-        socket.send(
-          new Writer()
-            .writeUInt8(0)
-            .writeString(token)
-            .writeUInt64([...stateId].reduce((r, v) => r * 36n + BigInt(parseInt(v, 36)), 0n))
-            .toBuffer()
-        );
-      socket.onmessage = ({ data }) => {
-        const reader = new Reader(new Uint8Array(data as ArrayBuffer));
-        const type = reader.readUInt8();
-        if (type === 0) {
-          const connection = new HathoraConnection(stateId, decodeStateSnapshot(reader), socket, onUpdate, onError);
-          resolve(connection);
-        } else {
-          reject("Unexpected message type: " + type);
-        }
-      };
-    });
+    const connection = new HathoraConnection(this.appId, stateId, transportType, onUpdate, onError);
+    await connection.connect(token);
```

Comment:
instead of this could we do something like return `HathoraConnection.connect(this.appId, stateId, transportType, onUpdate, onError)`? That way we can have a private `HathoraConnection` constructor which would guarantee that `state` is not undefined

Reply:
and we would be able to get rid of the `"Must wait on HathoraConnection.connect() before looking up state"` stuff

Reply:
Good call, much better approach

Reply:
discussed with @hpx7, we're going to leave as-is